.github/workflows/generate-docs.yml

```diff
@@ -1,11 +1,8 @@
 name: Generate and Deploy Documentation
 on:
-  push:
-    branches:
-      - main
-      - develop
-  release:
-    types: [published]
+  push:     
+    tags:
+      - "v*.*.*"
```

Comment:
I think it's still useful to run on `main` to have the "development" stream documented. It could help us check if doxygen has formatted everything as we expect, especially if we prepare for a minor or major update. 

We could call it `rolling` by moving to `doxygen/docs/versions/rolling`, and call it `latest` + `vX.Y.Z` on tag by making copies. For this you could re-use the "check if version changed" logic in build-release.yml. That would make the list of documented tags on https://aica-technology.github.io/control-libraries/ even nicer to navigate.